pkg/cmd/get.go

```diff
@@ -90,6 +94,27 @@ func (o *GetOptions) Complete(args []string) error {
 
 	o.RESTConfig = config
 
+	if o.Output != "" {
+		err = o.Validate()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *GetOptions) Validate() error {
+	validOutputFormats := []string{"json", "yaml"}
+	o.IsValidOutput = false
+	for _, validOutputFormat := range validOutputFormats {
+		if validOutputFormat == o.Output {
+			o.IsValidOutput = true
+		}
+	}
+	if !o.IsValidOutput {
+		return fmt.Errorf("unable to match a printer suitable for the output format %s, allowed formats are: json, yaml", o.Output)
+	}
```

Comment:
I think would be a good to move the definition of `validOutputFormats` outside of the `Validate` function to make it easier to find them. That would also let define them as enums which would let us use a switch case here and later on in get resources, which i think helps make the logic easier to follow.

```suggestion

type outputFormat string

func (of outputFormat) String() string {
    return string(of)
}

func (of outputFormat) Is(str string) bool {
    return string(of) == str
}

const (
    outputJson outputFormat = "json"
    outputYaml outputFormat = "yaml"
    outputTable outputFormat = "" // since table is the default
)

func (o *GetOptions) Validate() error {
	switch o.Output {
	case outputJson.String(), outputYaml.String(), outputTable.String():
	    return nil
	default:
	    return fmt.Errorf("unable to match a printer suitable for the output format %s, allowed formats are: json, yaml", o.Output)
	}
```

Reply:
If we use the flags already defined in k8s.io/cli-runtime/pkg/genericclioptions the same way that `kubectl get` does ([here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/get/get_flags.go#L155)) we can just remove this part as it's already defined there.